core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/operators/LogicalOp.scala

```diff
@@ -165,7 +165,7 @@ abstract class LogicalOp extends PortDescriptor with Serializable {
   private var context: WorkflowContext = _
 
   @JsonProperty(PropertyNameConstants.OPERATOR_ID)
-  private val operatorId: String = getClass.getSimpleName + "-" + UUID.randomUUID.toString
+  private var operatorId: String = getClass.getSimpleName + "-" + UUID.randomUUID.toString
```

Comment:
it's better to keep id unmutable. we can change the constructor of PhysicalOp to pass in an id optionally. That way, the id can only be assigned once at construction.

Reply:
I want the name of the materialization ops to be deterministic for each execution. This is a temporary workaround. I also tried to move the Id to the constructor but I feel we are unnecessarily complicating things.